Feature-IOPS-2995-EPS-Prescription-Tracker-Modeling-Prescriptions

KazeemHamzat opened the pull request.

KazeemHamzat committed:
Create Extension-England-Prescription-Pending-Cancellation

Create Extension-England-Prescription-Pending-Cancellation

KazeemHamzat committed:
Create Extension-DM-PrecscriberType

Create Extension-DM-PrecscriberType

KazeemHamzat committed:
Rename Extension-PendingCancellation

Rename Extension-PendingCancellation

KazeemHamzat committed:
Changing scope to element level

Changing scope to element level
RequestGroup.action.participant

KazeemHamzat committed:
Change scope to RequestGroup.action.

Change scope to RequestGroup.action.

KazeemHamzat committed:
Update Profiles:MedicationRequest & Dispense

Update Profiles:MedicationRequest & Dispense and associated extensions

KazeemHamzat committed:
Update to SD-RequestGroup &CS

Update to Capab. Stmt
Change RgGroup to UKCore
Change extension name Participant type

KazeemHamzat committed:
Reverting UKCore dependency to V0.03  pre-release

Reverting UKCore dependency to V0.03  pre-release

KazeemHamzat committed:
Revert changer to capab, Stms

Remove reference to RequestGroup profile

KazeemHamzat committed:
Rename Extension-ParticipantType

Rename Extension-ParticipantType

KazeemHamzat committed:
Chenge basedef to HL7

Chenge basedef to HL7

KazeemHamzat committed:
Change basedef to HL7

Change basedef to HL7

KazeemHamzat committed:
Create Example-Requestgroup-MedicationsPrescription-Example

Create Example-Requestgroup-MedicationsPrescription-Example

KazeemHamzat committed:
Testing Example

Testing Example

KazeemHamzat committed:
Correction to Extension URL

Correction to Extension URL

KazeemHamzat committed:
Change Example Code from Json to XML

Change Example Code from Json to XML

marcusfearnett-nhsdigital reviewed (dismissed):
Hi Kazeem been looking at your PR#455, and just don't have the background on the requirements to review. Of the reviewers on there, i think only Omar (who's away) or Naminder would have the squad 1 knowledge to do a proper job

Did notice
- copyright should be 2025+
- quite a few filenames have .StructureDefinition in them which we don't do
- we used to just do .xml files, some here are .json - do we care about this any more? I don't, but we should talk about it at DA before relaxing that rule

HamzatKazeem commented:
> Hi Kazeem been looking at your PR#455, and just don't have the background on the requirements to review. Of the reviewers on there, i think only Omar (who's away) or Naminder would have the squad 1 knowledge to do a proper job
> 
> Did notice
> 
> * copyright should be 2025+
> * quite a few filenames have .StructureDefinition in them which we don't do
> * we used to just do .xml files, some here are .json - do we care about this any more? I don't, but we should talk about it at DA before relaxing that rule

Thank for swift response Marcus. Let me provide some explanations if that will explain the context. 
Here are the new artefacts created.
1. EPS wants to use RequestGroup Resource to represents a medication prescription. Thus, the example, [Requestgroup-MedicationsPrescription-Example](https://github.com/NHSDigital/NHSEngland-FHIR-ImplementationGuide/blob/fabbb5c32affc8f0aff03af5634d2930b79ee505/RequestGroup/Requestgroup-MedicationsPrescription-Example.xml) has two MedicationRequest and 1 Medicationdispense as contained resources within it. 
2. RequestGroup.action represents some activities/staus iof prescription lifecyle(request and dispense). 
3. Extension-ParticipantType is backport of R5 RequestOrcestration.action.participant.typeReference.  This is needed to expand the options available in R4 RequestGroup.action.participant. Mainly used to identify the type of participant.
4. Extension-PendingCancellation was introduced to indicate either a prescription or any of its line item is pending cancellation. With RequestGroup as prescription, and MedicationRequest as line item.
5. England-RequestGroup as a profile.

**Other files**

1. NHSDigital-MedicationRequest: Added Extension-PendingCancellation to modify the meta to document the changes.
2. NHSDigital-MedicationDispense: Meta changes only.
3. Remaining Extensions: Meta changes and mapping of missing binding valuesets.
**File Extension.**
1. File with Json extension are expected to have it. E.g. package.json
2. Files with .Structuredefinition have always maintained their names but I can remove it if its required.

KazeemHamzat committed:
Update Copywright date to 2025

-Update Copywright date to 2025
-Revert to old in Package.json

KazeemHamzat committed:
Revert to old date in Package.json

Revert to old date in Package.json

HamzatKazeem commented:
> > Hi Kazeem been looking at your PR#455, and just don't have the background on the requirements to review. Of the reviewers on there, i think only Omar (who's away) or Naminder would have the squad 1 knowledge to do a proper job
> > Did notice
> > 
> > * copyright should be 2025+
> > * quite a few filenames have .StructureDefinition in them which we don't do
> > * we used to just do .xml files, some here are .json - do we care about this any more? I don't, but we should talk about it at DA before relaxing that rule
> 
> Thank for swift response Marcus. Let me provide some explanations if that will explain the context to reviewers. Here are the new artefacts created.
> 
> 1. EPS wants to use RequestGroup Resource to represent a medication prescription. Thus, the example, [Requestgroup-MedicationsPrescription-Example](https://github.com/NHSDigital/NHSEngland-FHIR-ImplementationGuide/blob/fabbb5c32affc8f0aff03af5634d2930b79ee505/RequestGroup/Requestgroup-MedicationsPrescription-Example.xml) has two MedicationRequest and 1 Medicationdispense as contained resources within it. Also have references to Provenance resources to document event history.
>    RequestGroup.action represents some activities/status of prescription lifecyle (request and dispense).
> 2. Extension-ParticipantType is backport of R5 RequestOrcestration.action.participant.typeReference. This is needed to expand the options available in R4 RequestGroup.action.participant. Mainly used to identify the type of participant.
> 3. Extension-PendingCancellation was introduced to indicate either a prescription or any of its line item is pending cancellation. With RequestGroup as prescription, and MedicationRequest as line item.
> 4. England-RequestGroup as a profile.
> 
> **Other files**
> 
> 1. NHSDigital-MedicationRequest: Added Extension-PendingCancellation and further modify the meta to document the changes.
> 2. NHSDigital-MedicationDispense: Meta changes only.
> 3. Remaining Extensions: Meta changes and mapping of missing binding ValueSets.
>    **File Extensions**
> 4. File with Json extension are expected to have it. E.g. package.json
> 5. Files with .Structuredefinition have always maintained their names but I can remove it if its required.

[Confluence page](https://nhsd-confluence.digital.nhs.uk/spaces/IOPS/pages/1002783529/IDA-018+-+Clinical+Prescription+Tracker+Modelling+of+Prescriptions)

DaveBarnet commented on RequestGroup/Requestgroup-MedicationsPrescription-Example.xml:
In the first <contained>, you need the resource name
<contained>
  <MedicationRequest>
    <intent value="order"/>
etc..
elements of the MedicationRequest in wronng order - as an example, the first contained resource should be
<MedicationRequest>
<id value='MedicationRequest-Prescription-Example'/>
    <extension url='https://fhir.nhs.uk/StructureDefinition/Extension-EPS-DispensingInformation'>
      <extension url='dispenseStatus'>
        <valueCoding>
          <system value='https://fhir.nhs.uk/CodeSystem/medicationdispense-type'/>
          <code value='0008'/>
          <display value='Item with dispenser'/>
        </valueCoding>
      </extension>
    </extension>
    <extension url='https://fhir.nhs.uk/StructureDefinition/Extension-PendingCancellation'>
      <valueBoolean value='true'/>
    </extension>
    <status value='active'/>
    <statusReason>
      <coding>
        <system value='https://fhir.nhs.uk/CodeSystem/medicationrequest-status-reason'/>
        <code value='0004'/>
        <display value='Clinical grounds'/>
      </coding>
    </statusReason>
    <intent value='order'/>
    <medicationCodeableConcept>
      <coding>
        <system value='http://snomed.info/sct'/>
        <code value='39720311000001101'/>
        <display value='Paracetamol 500mg soluble tablets'/>
      </coding>
    </medicationCodeableConcept>
    <subject>
      <reference value='Patient/Patient-StaceyTwitchett-Example'/>
    </subject>
    <dosageInstruction>
      <text value='Take one capsule three times a day'/>
    </dosageInstruction>
    <dispenseRequest>
      <quantity>
        <value value='30'/>
      </quantity>
    </dispenseRequest>
  </MedicationRequest>

Second MedicationRequest also elements in wrong order
  <MedicationRequest>
    <id value='MedicationRequest-PrescriptionUpdated-Example'/>
    <extension url='https://fhir.nhs.uk/StructureDefinition/Extension-EPS-DispensingInformation'>
      <extension url='dispenseStatus'>
        <valueCoding>
          <system value='https://fhir.nhs.uk/CodeSystem/medicationdispense-type'/>
          <code value='0002'/>
          <display value='Item not dispensed'/>
        </valueCoding>
      </extension>
    </extension>
    <status value='active'/>
      <intent value='order'/>
    <medicationCodeableConcept>
      <coding>
        <system value='http://snomed.info/sct'/>
        <code value='39720311000001101'/>
        <display value='Paracetamol 500mg soluble tablets'/>
      </coding>
    </medicationCodeableConcept>
    <subject>
      <reference value='Patient/Patient-StaceyTwitchett-Example'/>
    </subject>
    <dosageInstruction>
      <text value='Take two tablets every four hours as needed'/>
    </dosageInstruction>
    <dispenseRequest>
      <quantity>
        <value value='32'/>
      </quantity>
    </dispenseRequest>
    <eventHistory>
      <reference value='Provenance/Provenace-MedicationRequestHistory-Example'/>
    </eventHistory>
  </MedicationRequest>

Same with MedicationDispense (3rd contained) - should be something like
 <MedicationDispense>
    <id value='MedicationDispense-PendingPrescription-Example'/>
    <extension url='https://fhir.nhs.uk/StructureDefinition/Extension-DM-PrescriptionNonDispensingReason'>
      <valueCoding>
        <system value='https://fhir.nhs.uk/CodeSystem/medicationdispense-status-reason'/>
        <code value='0002'/>
        <display value='Clinically unsuitable'/>

      </valueCoding>
    </extension>
  <status value="completed"/>
    <medicationCodeableConcept>
      <coding>
        <system value='https://fhir.nhs.uk/CodeSystem/medication'/>
        <code value='789012'/>
        <display value='Paracetamol 500mg Tablets'/>

      </coding>
    </medicationCodeableConcept>
    <performer>
      <actor>
        <identifier>
          <system value='https://fhir.nhs.uk/Id/ods-organization-code'/>
          <value value='FCG71'/>
        </identifier>
      </actor>
    </performer>

    <type>
      <coding>
        <system value='https://fhir.nhs.uk/CodeSystem/medicationdispense-type'/>
        <code value='0002'/>
        <display value='Item not dispensed'/>

      </coding>
    </type>
    <authorizingPrescription>
      <reference value='#MedicationRequest-PrescriptionUpdated-Example'/>
    </authorizingPrescription>
    <quantity>
      <value value='32'/>
    </quantity>
    <dosageInstruction>
      <text value='Take two tablets every four hours as needed'/>
    </dosageInstruction>
  </MedicationDispense>

See email ...

KazeemHamzat replied:
Element within the resource instances of the example file have now been re-ordered.

DaveBarnet commented on StructureDefinition/England-RequestGroup.xml:
looks OK as far as I can tell

DaveBarnet reviewed (changes requested):
The structureDefs look OK as far as I can see, but the example needs some work as per the email I sent

KazeemHamzat committed:
Feedback Issues: Re-order  of elements

Feedback Issues: Re-order of elements

DaveBarnet reviewed (dismissed):
Example now validates - as far as I can see, you can pull the PR

KazeemHamzat committed:
Add CodeSystem & ValueSet PrescriptionTaskStatus

-Add CodeSystem & ValueSet PrescriptionTaskStatus
-Remove Structuredefinition from files names

KazeemHamzat committed:
Feedback Issues: Remodelled Example

Feedback Issues: Remodelled Example

KazeemHamzat committed:
update

KazeemHamzat committed:
Remodel Example

Remodel Example

KazeemHamzat committed:
Update Example

Update Example

KazeemHamzat committed:
Update Example

Update Example

KazeemHamzat committed:
Update Example: GroupIdentifier

Update Example: GroupIdentifier

KazeemHamzat committed:
Update Example: DispenseRequest

Update Example

KazeemHamzat committed:
Update MedicationDispense:Performer

Update MedicationDispense:Performer

KazeemHamzat committed:
Update Example

Update Example

KazeemHamzat committed:
Update \\example Telecom

Update \\example Telecom

KazeemHamzat committed:
MedicationDispense-PendingCancellation

-Typo in Surname

DaveBarnet reviewed (commented):
In one of the MedicationRequest examples in the extension Extension-EPS-DispensingInformation you've got
  <extension url="dispenseStatus">
   <valueCoding>
    <system value="https://fhir.nhs.uk/CodeSystem/medicationdispense-type" />
    **<code value="0008" />
    <display value="Item with dispenser" />**
   </valueCoding>
and in the other 
  <extension url="dispenseStatus">
   <valueCoding>
    <system value="https://fhir.nhs.uk/CodeSystem/medicationdispense-type" />
    **<code value="0007" />
    <display value="Item with dispenser" />**
   </valueCoding>
so a code / display need to change

should RequestGroup.status be "active"? rather than draft 

Other than that the example looks a lot better to me.

KazeemHamzat committed:
Feedback Issues:

- Change RequestGroup status to active
-Change MedicationRequest:extensionDispensingStatus for Repeat Prescription example to Item not Dispensed.

DaveBarnet reviewed (dismissed):
Looks good to me

omarisgreat commented on MedicationRequest/MedicationRequest-Prescription-Example.xml:
```diff
@@ -0,0 +1,86 @@
+<MedicationRequest xmlns="http://hl7.org/fhir">
+ <id value="MedicationRequest-Prescription-Example" />
+ <extension url="https://fhir.nhs.uk/StructureDefinition/Extension-PendingCancellation">
```
Replace this with TaskBusinessStatus if indicated by the EPS team

omarisgreat commented on MedicationRequest/MedicationRequest-RepeatPrescription-Example.xml:
```diff
@@ -0,0 +1,86 @@
+  </identifier>
+ </subject>
+ <requester>
+  <reference value="Organization/UKCore-Organization-HallgarthSurgery-Example" />
```
Remove from MedicationRequest and only represent this on RequestGroup

omarisgreat commented on StructureDefinition/NHSDigital-MedicationRequest.xml:
Remove max cardinality for TaskBusinessStatus extension
Remove min cardinality for requester

omarisgreat commented on RequestGroup/RequestGroup-MedicationsPrescription-Example.xml:
```diff
@@ -0,0 +1,189 @@
+    </extension>
+  </extension>
+  <extension url="https://fhir.nhs.uk/StructureDefinition/Extension-PendingCancellation">
+    <valueBoolean value="false" />
```
This could possibly be replaced by TaskBusinessStatus extension

omarisgreat commented on RequestGroup/RequestGroup-MedicationsPrescription-Example.xml:
```diff
@@ -0,0 +1,189 @@
+  </subject>
+  <authoredOn value="2025-01-29T12:00:00Z" />
+  <author>
+    <reference value="Organization/UKCore-Organization-HallgarthSurgery-Example" />
```
Need to check with EPS team whether this should be a practitioner/practitionerRole instead

KazeemHamzat replied:
Changed to Practitioner

omarisgreat commented on RequestGroup/RequestGroup-MedicationsPrescription-Example.xml:
```diff
@@ -0,0 +1,189 @@
+    </identifier>
+  </author>
+  <action>
+    <title value="Prescription status transitions" />
```
Rearrange actions so that we have 1-4 MedicationRequest (line item) actions, and 1 'Status Transitions' actions, which cover all changes/messages related to a prescription, e.g. release/dispense etc.

omarisgreat commented on StructureDefinition/England-RequestGroup.xml:
```diff
@@ -0,0 +1,91 @@
+  <baseDefinition value="http://hl7.org/fhir/StructureDefinition/RequestGroup" />
+  <derivation value="constraint" />
+  <differential>
+    <element id="RequestGroup.extension:pendingCancellation">
```
Can remove this

omarisgreat commented on StructureDefinition/Extension-PendingCancellation.xml:
Potentially remove

omarisgreat reviewed (changes requested):
remove https://fhir.nhs.uk/CodeSystem/EPS-task-business-status canonical URL from DMmedicationdispenseprescriptionstatusns naming system.

Changes mostly good, just change the RequestGroup example to have MedicationRequests separate from status transitions. 

Also please, create a Bundle with all the referenced resources to represent a response from the clinical prescription tracker e.g.

Bundle
- RequestGroup
-- MedicationRequest (1..4) (line items)
-- MedicationDispense (0..*) (status transitions)
-- Patient (1..1) (subject)
-- PractitionerRole (1..1) (author)
-- Organization (0..*) (participants)

KazeemHamzat committed:
Merge branch 'nhsdigital_main' of https://github.com/NHSDigital/NHSEngland-FHIR-ImplementationGuide into Feature-IOPS-2995A-EPS-Update-CodeSystem-NPPTS-Status

KazeemHamzat committed:
Merge branch 'Feature-IOPS-2995-EPS-Prescription-Tracker-Modeling-Prescriptions' into Feature-IOPS-2995A-EPS-Update-CodeSystem-NPPTS-Status

KazeemHamzat committed:
Feedback Issues

-Remove Extension-pwdingCancellation and replaced with Task Business Status
-Remove max Cardinality of 1 ffor MedicationDispense.performer & -Extension-Task-BusinessStatus

KazeemHamzat committed:
Feedback Issues: Remodel Example and Added Bundle

Feedback Issues: Remodel Example and Added Bundle

KazeemHamzat committed:
Feedback Issues: Added Bundle

Feedback Issues

KazeemHamzat committed:
Feedback Issues: Added Bundle

Feedback Issues: Added Bundle

KazeemHamzat committed:
Update Bundle

Update Bundle

KazeemHamzat committed:
Update Bundle

Update Bundle

KazeemHamzat committed:
Update Bundle

Update Bundle

KazeemHamzat committed:
Update Bundle-1

Update Bundle 1

KazeemHamzat committed:
UpdateBundle: Fix MedicationDispense URL

UpdateBundle: Fix MedicationDispense URL

KazeemHamzat committed:
Update Bundle

Update Bundle

KazeemHamzat committed:
Removing the contained Restriction

Removing the contained Restriction

KazeemHamzat committed:
Remove Bundle

KazeemHamzat committed:
Update MedicationDispense

Update MedicationDispense

KazeemHamzat committed:
Update MDispense & Bundle

Remove Identifier from MedicationDispense.Organization

omarisgreat commented on StructureDefinition/NHSDigital-MedicationDispense.xml:
You might need to leave this as is as removal of the prescriptionStatus extension would break the existing EPS implementation. Could you instead just remove the max cardinality on prescriptionStatus?

omarisgreat commented on StructureDefinition/Extension-DM-PrescriptionStatusHistory.xml:
```diff
@@ -2,7 +2,7 @@
   <id value="Extension-DM-PrescriptionStatusHistory" />
   <url value="https://fhir.nhs.uk/StructureDefinition/Extension-DM-PrescriptionStatusHistory" />
-  <version value="0.2.0" />
+  <version value="An extension use to record the status of a prescription. The data expected to be collected with this extension is a subset value in &lt; a href=&quot;https://simplifier.net/nhsdigital/cb27c46c-ea79-4609-8a79-efa1e799755c&quot; target=&quot;_blank&quot;&gt;ExtensionDMPrescriptionStatusHistory&lt;/a&gt;, and the 2 extensions both reference the same &lt; a href=&quot;https://simpli ...
```
This looks like it should be in a description rather than version field

omarisgreat commented on StructureDefinition/England-RequestGroup.xml:
```diff
@@ -27,26 +27,6 @@
-      </type>
-      <isModifier value="false" />
-    </element>
-    <element id="RequestGroup.extension:prescriptionTaskStatusReason">
```
Wondering why this extension was removed, we need to record the status of a prescription on the RequestGroup, and status of line items on MedicationRequests, as in the RequestGroup example

omarisgreat reviewed (commented):
Couple of questions regarding changes to extensions

KazeemHamzat committed:
Feedback Issues

Feddback Issues

KazeemHamzat committed:
Merge branch 'Feature-IOPS-2995-EPS-Prescription-Tracker-Modeling-Prescriptions' of https://github.com/NHSDigital/NHSEngland-FHIR-ImplementationGuide into Feature-IOPS-2995A-EPS-Update-CodeSystem-NPPTS-Status

KazeemHamzat committed:
Update Bundle

Update Bundle

KazeemHamzat committed:
Update ValueSet/DM-medicationdispense-type

Update ValueSet/DM-medicationdispense-type

MatthewPopat-NHS commented on RequestGroup/RequestGroup-MedicationsPrescription-Example.xml:
```diff
@@ -0,0 +1,206 @@
+      </valueCoding>
+    </extension>
+  </extension>
+  <extension url="https://fhir.nhs.uk/StructureDefinition/Extension-DM-PrescriptionStatusHistory">
```
It doesn't make sense to have multiple of these, the prescription only has one statuse.

omarisgreat replied:
What about the case where a prescription is with dispenser but is also pending cancellation (as they are currently represented within the same codesystem, we would require 2 instances of the extension if they can both be applicable at the same time)

MatthewPopat-NHS replied:
ah yes I see the confusion, the 9005 "Pending Cancellation" status doesn't actually mean that the prescription has a pending cancellation against it, it means that it was a pending (future dated) prescription that has been cancelled. When a cancellation comes in that can't be immediately actioned the status doesn't change, the pending cancellation gets recorded in a separate field, hence me suggesting a new extension.

omarisgreat replied:
Ok, we can reinstate the pending cancellation extension, this was taken out in review due to the perceived overlap between this an 9005. @KazeemHamzat, can you action?

MatthewPopat-NHS commented on RequestGroup/RequestGroup-MedicationsPrescription-Example.xml:
```diff
@@ -0,0 +1,206 @@
+    <reference value="PractitionerRole/PractitionerRole-LucyHale-Example" />
+    <identifier>
+      <use value="official" />
+      <system value="https://fhir.hl7.org.uk/Id/gmc-number" />
```
We won't have the gmc-number or any professional identifier available. We were hoping to just use the prescribing organisation ODS code here.

omarisgreat replied:
Isn't the prescriber a particular individual (as per the EPS spec?), will there not be a reference to the individual within the prescription tracker?

MatthewPopat-NHS replied:
The prescriber is a specific individual but we aren't showing that information in tracker and the data source we are using on Spine only exposes the ODS code.

omarisgreat replied:
Acknowledged, @KazeemHamzat, could you replace the reference with an Organization reference and ODS code for the requesting organisation?

MatthewPopat-NHS commented on RequestGroup/RequestGroup-MedicationsPrescription-Example.xml:
```diff
@@ -0,0 +1,206 @@
+      <value value="C9999999" />
+    </identifier>
+  </author>
+  <action id="All-MedicationsID">
```
I don't think we need a separate "all medications" action, because all the medication requests would always be present on the "Prescription upload successful" action.

omarisgreat replied:
The thinking behind this was that the prescription would exist outside the actions that were performed on the prescription so, in this case, the prescription upload successful event itself could be stripped down to just details about the event, rather than referencing the actual medications. Would that work?

MatthewPopat-NHS replied:
Yeah that makes sense but in that case we should probably also move the repeat timing information here so that all the timings on the other actions are homogeneous.

omarisgreat replied:
Agreed, @KazeemHamzat, could you look at moving the repeat timings into the RequestGroup itself or the action for all medications?

MatthewPopat-NHS commented on RequestGroup/RequestGroup-MedicationsPrescription-Example.xml:
```diff
@@ -0,0 +1,206 @@
+      </resource>
+    </action>
+    <action id="PrescriptionStatus-MedicationNumber-2">
+      <title value="Medication-Number-2 upload successful" />
```
you can't add medication requests to a prescription that already exists.

omarisgreat replied:
This can be removed @KazeemHamzat

MatthewPopat-NHS commented on RequestGroup/RequestGroup-MedicationsPrescription-Example.xml:
```diff
@@ -0,0 +1,206 @@
+          <valueReference>
+            <identifier>
+              <system value="https://fhir.nhs.uk/Id/ods-organization-code" />
+              <value value="A83008" />
```
can we not put this identifier directly as the participant rather than an extension?

omarisgreat replied:
Unfortunately, the R4 version does not allow organisation as a target on participant so this needed to be backported from a later version of FHIR

MatthewPopat-NHS commented on RequestGroup/RequestGroup-MedicationsPrescription-Example.xml:
```diff
@@ -0,0 +1,206 @@
+        </extension>
+      </participant>
+      <resource>
+        <reference value="MedicationRequest/MedicationRequest-Prescription-Example" />
```
can the MedicationRequest resources not be contained within this RequestGroup?

omarisgreat replied:
Guidance has been obtained from IOPS that resources being references are preferred to 'contained resources' as these should only be used in particular circumstances (which do not apply to EPS/prescription tracker use cases: https://www.hl7.org/fhir/references.html#contained)

MatthewPopat-NHS replied:
"does not have an independent existence apart from the resource that contains it" seem like it is true for the MedicationRequests, there's no way of interacting with an individual MedicationRequest outside of the prescription that contains it.

omarisgreat replied:
OK, we can take this back to IOPS to review the decision, we can probably switch back to contained resources based on the information you provided. @KazeemHamzat, could you draft something for the next team meeting and look at editing the example to represent what using contained resources for the line items would look like?

MatthewPopat-NHS commented on RequestGroup/RequestGroup-MedicationsPrescription-Example.xml:
```diff
@@ -0,0 +1,206 @@
+        </extension>
+      </participant>
+      <resource>
+        <reference value="MedicationRequest/MedicationRequest-RepeatPrescription-Example" />
```
This should be a MedicationDispense not a MedicationRequest.

omarisgreat replied:
This can be changed @KazeemHamzat

MatthewPopat-NHS commented on RequestGroup/RequestGroup-MedicationsPrescription-Example.xml:
```diff
@@ -0,0 +1,206 @@
+        </extension>
+      </participant>
+      <resource>
+        <reference value="MedicationRequest/MedicationRequest-Prescription-Example" />
```
we wouldn't include any resource on the release request message, the release always applies to the whole prescription.

omarisgreat replied:
This can be removed @KazeemHamzat

MatthewPopat-NHS commented on RequestGroup/RequestGroup-MedicationsPrescription-Example.xml:
```diff
@@ -0,0 +1,206 @@
+        <coding>
+          <system value="https://fhir.nhs.uk/CodeSystem/EPS-task-business-status" />
+          <code value="0002" />
+          <display value="Item not dispensed" />
```
"Item not dispensed" isn't a valid "EPS-task-business-status" (prescription status). The example I used of "0003" "With Dispenser - Active" would be better.

omarisgreat replied:
This can be changed @KazeemHamzat
